src/PKSim.Core/Services/PKAnalysesTask.cs

```diff
@@ -430,23 +523,97 @@ public void CalculateBioavailabilityFor(Simulation simulation, string compoundNa
-         simulation.CompoundPKFor(compoundName).CmaxDDI = pkParameterInBloodCurveFor(ddiRatioSimulation, compoundName, cmaxPKParameterName);
+
+         simulation.CompoundNames.Each(compoundName =>
+         {
+            var compound = simulation.Compounds.FindByName(compoundName);
+            var applicationType = applicationTypeFor(simulation, compound);
+
+            var aucPKParameterName = pkParameterNameForAUCRatio(applicationType);
+            var cmaxPKParameterName = pkParameterNameForCmaxRatio(applicationType);
+
+            if (simulation is PopulationSimulation populationSimulation)
+            {
+               // We know this is a valid cast because it is cloned from simulation
+               var ddiPopulationSimulation = ddiRatioSimulation.DowncastTo<PopulationSimulation>();
+
+               var ddiCompoundPKContext = new CompoundPKContext();
+
+               // First calculate the compound context. We will know the AucInf and CMax when DDI is turned off
+               CalculateFor(ddiPopulationSimulation, ddiCompoundPKContext);
+               var parameterNames = new[] { CoreConstants.PKAnalysis.AUCRatio, CoreConstants.PKAnalysis.C_maxRatio };
+
+               // From the ddi context, create the context from the original simulation and the ddi simulation for calculating ddi ratio parameters
+               var contextForPopulationSimulation = createContextForPopulationSimulation(ddiCompoundPKContext, mapFromDDISimulationCompoundPK, compoundName, populationSimulation, parameterNames);
+
+               // Calculate all the parameters again, including now - ddi ratio parameters from the context
+               populationSimulation.PKAnalyses = CalculateFor(populationSimulation, contextForPopulationSimulation);
+            }
+            else if (simulation is IndividualSimulation individualSimulation)
+            {
+               individualSimulation.AucDDI[compoundName] = pkParameterInBloodCurveFor(ddiRatioSimulation, compoundName, aucPKParameterName);
+               individualSimulation.CMaxDDI[compoundName] = pkParameterInBloodCurveFor(ddiRatioSimulation, compoundName, cmaxPKParameterName);
+            }
+         });
+         simulation.HasChanged = true;
+
+      }
+
+      private CompoundPKContext createContextForPopulationSimulation(CompoundPKContext contextCompoundPK, Func<CompoundPK, CompoundPK> compoundPKMapper, string compoundName, PopulationSimulation populationSimulation, string[] parameterNames)
+      {
+         var contextForPopulationSimulation = new CompoundPKContext();
+         contextForPopulationSimulation.AddCompoundPK(compoundPKMapper(contextCompoundPK.CompoundPKFor(compoundName)));
+         contextForPopulationSimulation.InitializeQuantityPKParametersFrom(populationSimulation);
+         parameterNames.Each(parameterName => { removeQuantityPKParameterFromCompound(compoundName, contextForPopulationSimulation, parameterName); });
```

Comment:
Then after that, we have to remove the ratio parameters that are being calculated